src/runtime/refiner.ts

```diff
@@ -848,7 +911,7 @@ export class RefinementOperator {
   }
 
   eval(exprs: ExpressionPrimitives[]): ExpressionPrimitives {
-    return this.opInfo.evalFn(exprs);
+    return evalTable[this.op](exprs);
```

Comment:
`evalTable[this.op]` could be undefined. If we changed the `evalTable`'s type to be more specific than `Dictionary`, I think the compiler would be able to track the values for us. Might be worth leaving for now but coming back to when we have a little time.